README.md

```diff
@@ -264,6 +264,125 @@ safe_create_partitioned_table :table, type: :range, key: :example_column, infer_
+
+Additionally, this method allows you to configure a subset of attributes on the record stored in the [part\_config](https://github.com/pgpartman/pg_partman/blob/master/doc/pg_partman.md#tables) table.
+These options are delegated to the `unsafe_partman_update_config` method to update the record:
+
+- `infinite_time_partitions`. Partman defaults this to `false` but we default to `true`
+- `inherit_privileges`. Partman defaults this to `false` but we default to `true`
+- `retention`. Partman defaults this to `null`
+- `retention_keep_table`. Partman defaults this to `true`
+
+With only the required args:
+
+```ruby
+safe_create_partitioned_table :table, type: :range, key: :created_at do |t|
+  t.timestamps :null => false
+end
+
+safe_partman_create_parent :table, key: :created_at, interval: "weekly"
+```
+
+With custom overrides:
+
+```ruby
+safe_create_partitioned_table :table, type: :range, key: :created_at do |t|
+  t.timestamps :null => false
+  t.text :some_column
+end
+
+# Partman will reference the template table to create unique indexes on child tables
+safe_create_table :table_template, id: false do |t|
+  t.text :some_column, index: { unique: true }
+end
+
+safe_partman_create_parent :table,
+  key: :created_at,
+  interval: "weekly",
+  template_table: :table_template,
+  premake: 10,
+  start_partition: (Time.current + 1.month).to_fs(:db),
+  infinite_time_partitions: false,
+  inherit_privileges: false
+```
+
+#### unsafe\_partman\_create\_parent
+
+We have chosen to flag the use of `retention` and `retention_keep_table` as an unsafe operation.
+While we recognize that these options are useful, we think they fit in the same category as `drop_table` and `rename_table`, and are therefore unsafe from an application perspective.
+If you wish to define these options, you must use this method.
+
+```ruby
+safe_create_partitioned_table :table, type: :range, key: :created_at do |t|
+  t.timestamps :null => false
+end
+
+unsafe_partman_create_parent :table,
+  key: :created_at,
+  interval: "weekly",
+  retention: "60 days",
+  retention_keep_table: false
+```
+
+#### safe\_partman\_update\_config
+
+There are some partitioning options that cannot be set in the call to `create_parent` and are only available in the `part_config` table.
+As mentioned previously, you can specify these args in the call to `safe_partman_create_parent` or `unsafe_partman_create_parent` which will be delegated to this method.
+Calling this method directly will be useful if you need to modify your partitioned table after the fact.
+
+Allowed keyword args:
+
+- `infinite_time_partitions`
+- `inherit_privileges`
+- `premake`
+- `retention`
+- `retention_keep_table`
+
+Note that we detect if the value of `inherit_privileges` is changing and will automatically call `safe_partman_reapply_privileges` to ensure permissions are propagated to existing child partitions.
```

Comment:
Same comment as above. Is this too magical?

Reply:
No, I think this is just eliminating a sharp edge from partman.